core/utils/mailbox_test.go

```diff
@@ -30,17 +42,13 @@ func TestMailbox(t *testing.T) {
 		}
 	}()
 
-	for _, i := range toDeliver {
-		m.Deliver(i)
-	}
 	close(m.Notify())
-
 	<-chDone
 
-	if len(recvd) > 10 {
-		t.Fatal("received too many")
-	} else if len(recvd) < 10 {
-		t.Fatal("received too few")
+	if len(recvd) > capacity {
+		t.Fatal("received too many:", recvd)
```

Comment:
nit: whole block of code 48-51 seems redundant since line 53 will anyways check it all, and also give the right error, in terms of what was different than expected.